truth_set/java/structure/organisation/statement-issues/OrderProcessor.java

```diff
@@ -0,0 +1,79 @@
+import java.util.List;
+import java.util.Arrays;
+
+class Item {
+    private String name;
+    private double price;
+    private int quantity;
+
+    public Item(String name, double price, int quantity) {
+        this.name = name;
+        this.price = price;
+        this.quantity = quantity;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+}
+
+class Order {
+    private String id;
+    private List<Item> items;
+    private String discountCode;
+
+    public Order(String id, List<Item> items, String discountCode) {
+        this.id = id;
+        this.items = items;
+        this.discountCode = discountCode;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public List<Item> getItems() {
+        return items;
+    }
+
+    public String getDiscountCode() {
+        return discountCode;
+    }
+}
+
+public class OrderProcessor {
+    public static double processOrder(Order order, double taxRate) {
+        double total = (calculateItemsTotal(order.getItems()) - 
+                      (order.getDiscountCode() != null ? 50 : 0)) * 
+                      (1 + taxRate);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting tax calculation logic</b></div>
<div id="fix">

Consider extracting the tax calculation logic into a separate method. The calculation `(1 + taxRate)` could be moved to a dedicated method like `calculateTaxAmount()` to improve code readability and maintainability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -49,9 +49,13 @@
     public static double processOrder(Order order, double taxRate) {
 -        double total = (calculateItemsTotal(order.getItems()) -
 -                      (order.getDiscountCode() != null ? 50 : 0)) *
 -                      (1 + taxRate);
 +        double subtotal = calculateItemsTotal(order.getItems()) -
 +                         (order.getDiscountCode() != null ? 50 : 0);
 +        double total = calculateTaxAmount(subtotal, taxRate);
 +    }
 +
 +    private static double calculateTaxAmount(double amount, double taxRate) {
 +        return amount * (1 + taxRate);
     }
```




```
 @@ -49,9 +49,13 @@
     public static double processOrder(Order order, double taxRate) {
 -        double total = (calculateItemsTotal(order.getItems()) -
 -                      (order.getDiscountCode() != null ? 50 : 0)) *
 -                      (1 + taxRate);
 +        double subtotal = calculateItemsTotal(order.getItems()) -
 +                         (order.getDiscountCode() != null ? 50 : 0);
 +        double total = calculateTaxAmount(subtotal, taxRate);
 +    }
 +
 +    private static double calculateTaxAmount(double amount, double taxRate) {
 +        return amount * (1 + taxRate);
     }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2522127787>#dcd59f</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged